### PR TITLE
Nightly Slurm CI Rocky update workflow

### PR DESCRIPTION
**Pull request by bertiethorpe**
- New workflow pre-builds ofed and cuda packages into "latest" images, nightly.
- "Latest" images are synced across CI clouds.
- Normal fatimage workflow then uses these "latest" nightly images as new source images.
- Trivy scan adjusted to run on the nightly base image builds, and on merge to main in separate workflow.

### CODE REVIEW

**Comment by bertiethorpe** [comment]
Successful nightly build and image sharing between clouds:

https://github.com/stackhpc/ansible-slurm-appliance/actions/runs/11106417297

**Comment by bertiethorpe** [comment]
Successful nightly build and image sharing between clouds:

https://github.com/stackhpc/ansible-slurm-appliance/actions/runs/11129155877/job/30928488462

**Comment by bertiethorpe** [comment]
First successful new fatimage build on SMS:
https://github.com/stackhpc/ansible-slurm-appliance/actions/runs/11131571980

RL8 - 1h 19m 8s
RL9 - 1h 9m 23s
RL9-cuda - 1h 34m 40s

**Comment by bertiethorpe** [comment]
new fatimage build on Leafcloud:
https://github.com/stackhpc/ansible-slurm-appliance/actions/runs/11136671437

RL8 - 49m 33s
RL9 - 39m 16s
RL9-cuda - 1h 9m 31s

**Comment by bertiethorpe** [comment]
Fatimage leafcloud build after moving trivy scan to separate workflow:
https://github.com/stackhpc/ansible-slurm-appliance/actions/runs/11162164400

Run time: 35m

**Comment by bertiethorpe** [comment]
Trivy scan on push works:
https://github.com/stackhpc/ansible-slurm-appliance/actions/runs/11177947378

**Review by sjpb — CHANGES_REQUESTED** [review]
Some minor nits.

File: .github/workflows/fatimage.yml [hunk 1]
```diff
@@ -66,15 +68,28 @@ jobs:
...
         id: packer_build
         run: |
+          set -x
           . venv/bin/activate
           . environments/.stackhpc/activate
           cd packer/
           packer init .
-          PACKER_LOG=1 packer build -on-error=${{ vars.PACKER_ON_ERROR }} -only=${{ matrix.build }} -var-file=$PKR_VAR_environment_root/${{ env.CI_CLOUD }}.pkrvars.hcl openstack.pkr.hcl
+
+          if [[ "${{ matrix.build }}" == *"cuda"* ]]; then
```
[the remark this answers is not in the document]
Comment (sjpb) [reply .github/workflows/fatimage.yml:81]: TBD - am going to provide another response on this.

File: ansible/bootstrap.yml [hunk 2]
```diff
@@ -148,9 +148,9 @@
   tags: cockpit
   tasks:
     - name: Remove RHEL cockpit
-      dnf:
-        name: cockpit-ws
-        state: "{{ appliances_cockpit_state }}"
+      command: dnf -y remove cockpit-ws
```
[the remark this answers is not in the document]
Comment (sjpb) [reply ansible/bootstrap.yml:151]: From @bertiethorpe:

> This was in the PR from when random packer build tasks got a lot slower. This was a suggestion from Matt A if I remember correctly. I can't remember if it was this in the end, but packer builds starting speeding up again after I did it

[the remark this answers is not in the document]
Comment (sjpb) [reply ansible/bootstrap.yml:151]: Ok could you just add a comment in to that effect then pls.

**Comment by bertiethorpe** [comment]
fat image build:
https://github.com/stackhpc/ansible-slurm-appliance/actions/runs/11253430727

File: .github/workflows/fatimage.yml [hunk 3]
```diff
@@ -66,15 +78,24 @@ jobs:
...
         id: packer_build
         run: |
+          set -x
           . venv/bin/activate
           . environments/.stackhpc/activate
           cd packer/
           packer init .
-          PACKER_LOG=1 packer build -on-error=${{ vars.PACKER_ON_ERROR }} -only=${{ matrix.build }} -var-file=$PKR_VAR_environment_root/${{ env.CI_CLOUD }}.pkrvars.hcl openstack.pkr.hcl
+
+          SOURCE_IMAGE=${{ fromJSON(env.SOURCE_IMAGES_MAP)[matrix.os_version][matrix.build] }}
```
Comment (sjpb) [inline .github/workflows/fatimage.yml:91]: Would this not be cleaner in the task's env? If its not at the top-level env.

File: .github/workflows/nightlybuild.yml [hunk 4]
```diff
@@ -0,0 +1,266 @@
...
+      - name: Build fat image with packer
+        id: packer_build
+        run: |
+          set -x
+          . venv/bin/activate
+          . environments/.stackhpc/activate
+          cd packer/
+          packer init .
+
+          SOURCE_IMAGE=${{ fromJSON(env.SOURCE_IMAGES_MAP)[matrix.os_version] }}
```
Comment (sjpb) [inline .github/workflows/nightlybuild.yml:88]: As for same in fatimage.

File: packer/openstack.pkr.hcl [hunk 5]
```diff
@@ -47,12 +47,8 @@ variable "os_version" {
...
-  description = "name of source image, keyed from var.os_version"
-  default = {
-    RL8: "Rocky-8-GenericCloud-Base-8.9-20231119.0.x86_64.qcow2"
-    RL9: "Rocky-9-GenericCloud-Base-9.4-20240523.0.x86_64.qcow2"
-  }
+  type = string
+  description = "name of source image"
 }
 
 variable "source_image" {
```
Comment (sjpb) [inline packer/openstack.pkr.hcl:54]: This needs to NOT be a map any more, same as source_image_name. Should still have default of null though.

**Review by sjpb — CHANGES_REQUESTED** [review]

**Comment by bertiethorpe** [comment]
https://github.com/stackhpc/ansible-slurm-appliance/actions/runs/11258069062

**Review by sjpb — APPROVED** [review]
LGTM

**Pull request merged**